### DISCUSSION TITLE
pipx install eth-brownie , causing error

### DISCUSSION THREAD

**Discussion by km123abca** [smartcontractkit/full-blockchain-solidity-course-py#1162] [Q&A] [answered]
pipx install eth-brownie  is causing an error. The end of the error stack is as follows

\lib\encodings\cp1252.py", line 19, in encode
    return codecs.charmap_encode(input,self.errors,encoding_table)[0]
UnicodeEncodeError: 'charmap' codec can't encode character '\u2502' in position 559: character maps to <undefined>

How do I resolve this? Please help

**Comment by cromewar** [comment, marked as answer]
This is a problem with your python environment, so please do the following.

1. Install python version 3.8
2. create a virtual env wit that version
3. install brownie using pip instead of pipx

**Comment by n4n0b1t3** [comment]
Just listen to @cromewar 
Here is an [updated article on how to do that](https://github.com/smartcontractkit/full-blockchain-solidity-course-py/discussions/1109).

**Reply by mart-cervants** [reply]
When I tried to install eth-browniew using pipx I got this:

The virtual environment was not created successfully because ensurepip is not
available.  On Debian/Ubuntu systems, you need to install the python3-venv
package using the following command.

    apt-get install python3-venv

You may need to use sudo with that command.  After installing the python3-venv
package, recreate your virtual environment.

Failing command: ['/home/martcerv/.local/pipx/shared/bin/python3', '-Im', 'ensurepip', '--upgrade', '--default-pip']


'/usr/bin/python3 -m venv --clear /home/martcerv/.local/pipx/shared' failed


I also follow [this](https://stackoverflow.com/questions/70828854/cant-install-eth-brownie-on-ubuntu) but it doesn't work, any suggestion?

I'm using Ubuntu 18.04 in WSL

**Reply by cromewar** [reply]
Just install using regular pip instead of pipx, and is not recommend to install using sudo as it represents a security vulnerability.
